cluster/src/shard_lock_manager.rs

```diff
@@ -344,50 +491,104 @@ impl ShardLock {
         OnExpired: FnOnce(ShardId) -> Fut + Send + 'static,
         Fut: Future<Output = ()> + Send + 'static,
     {
-        let mut lease = Lease::new(lease_id, Duration::from_secs(self.ttl_sec));
-        let (lease_expire_notifier, mut lease_expire_receiver) = oneshot::channel();
-        let (keepalive_stop_sender, keepalive_stop_receiver) = oneshot::channel();
-        lease
-            .start_keepalive(
-                keepalive_stop_receiver,
-                lease_expire_notifier,
-                etcd_client,
-                runtime,
-            )
-            .await?;
-
-        let lock_expire_check_interval = LOCK_EXPIRE_CHECK_INTERVAL;
+        // Try to acquire the lock to ensure there is only one running keepalive
+        // procedure.
+        let initial_state = LeaseState::new(expired_at);
+        let lease = Arc::new(Lease::new(
+            lease_id,
+            Duration::from_secs(self.ttl_sec),
+            initial_state,
+        ));
+
         let shard_id = self.shard_id;
+        let rpc_timeout = self.rpc_timeout;
```

Comment:
The network jitter tolerating startegy here seems complex, maybe add a detail comment to describe the general procedure?